### PR TITLE
Make sure to delete RelayIds when deleting nodes

### PR DESCRIPTION
**Pull request by do4gr**
- fixes it for nested deletes and deleteMany
- also cleans up DatabaseMutationbuilder
- also removes the remains of FieldMirrorColumns from the code

### CODE REVIEW

File: server/api/src/main/scala/com/prisma/api/database/DatabaseApiTestDatabaseMutationBuilder.scala [hunk 1]
```diff
@@ -0,0 +1,123 @@
+package com.prisma.api.database
+
+import com.prisma.shared.models.TypeIdentifier.TypeIdentifier
+import com.prisma.shared.models.{Model, TypeIdentifier}
+import slick.dbio.DBIOAction
+import slick.jdbc.MySQLProfile.api._
+
+object DatabaseApiTestDatabaseMutationBuilder {
```
Comment (mavilein) [inline server/api/src/main/scala/com/prisma/api/database/DatabaseApiTestDatabaseMutationBuilder.scala]: If this is indeed just used in tests, it should reside in `src/test/scala`

File: server/api/src/main/scala/com/prisma/api/database/mutactions/mutactions/DeleteDataItemNested.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import com.prisma.api.mutations.NodeSelector
+import com.prisma.api.schema.APIErrors
+import com.prisma.shared.models.Project
+import slick.jdbc.MySQLProfile.api._
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.util.{Failure, Success, Try}
+
+case class DeleteDataItemNested(project: Project, where: NodeSelector) extends ClientSqlDataChangeMutaction {
```
Comment (mavilein) [inline server/api/src/main/scala/com/prisma/api/database/mutactions/mutactions/DeleteDataItemNested.scala]: Could we directly substitute the uses of `DeleteDataItem` with this one? This one looks better to me.

File: server/api/src/main/scala/com/prisma/api/database/DatabaseApiTestDatabaseMutationBuilder.scala (hunk 1 above)
Comment (do4gr) [reply server/api/src/main/scala/com/prisma/api/database/DatabaseApiTestDatabaseMutationBuilder.scala]: moved it

File: server/api/src/main/scala/com/prisma/api/database/mutactions/mutactions/DeleteDataItemNested.scala (hunk 2 above)
Comment (do4gr) [reply server/api/src/main/scala/com/prisma/api/database/mutactions/mutactions/DeleteDataItemNested.scala]: We can probably do that at a later point when we do not discern between top-level and nested mutactions that heavily anymore. At the moment that change would require a lot of extra refactoring.

**Pull request merged**